src/Storefront/Resources/views/components/Sw/Content/Modal.js

```diff
@@ -0,0 +1,93 @@
+class Modal extends window.ShopwareComponent {
+
+    static selector = '[data-component="Modal"]';
+
+    static options = {
+        ajaxUrl: null,
+        setTitleFromAjaxContent: false,
+    };
+
+    init() {
+        this.modalBody = this.el.querySelector('.sw-modal__body');
+        this.modalTitle = this.el.querySelector('.sw-modal__title');
+
+        this.registerRemoteModalEvents();
+    }
+
+    registerRemoteModalEvents() {
+        if (!this.options.ajaxUrl) {
+            return;
+        }
+
+        this.el.addEventListener('show.bs.modal', this.addLoadingState.bind(this));
+        this.el.addEventListener('shown.bs.modal', this.fetchContent.bind(this));
+    }
+
+    addLoadingState() {
+        if (this.options.setTitleFromAjaxContent) {
+            this.modalTitle.classList.add('placeholder-glow', 'w-25');
+            this.modalTitle.innerHTML = '<span class="placeholder col-12"></span>';
+        }
+
+        this.modalBody.innerHTML = this.getLoaderTemplate();
+    }
+
+    fetchContent() {
+        fetch(this.options.ajaxUrl)
+            .then(response => response.text())
+            .then(content => this.renderContent(content))
+            .catch(error => this.handleError(error))
+    }
+
+    renderContent(content) {
+        this.modalBody.innerHTML = content;
```

Comment:
We could add more events here, to make it extendable in some ways.

```suggestion
({ content } = Shopware.emitInterception(`${this.componentName}:PreRenderContent`, { content }));

 this.modalBody.innerHTML = content;
```